Makefile

```diff
@@ -16,3 +16,9 @@ run:
 test: ruff pylint
 	export PYTHONPATH=src/main/py; \
 	python3 -m unittest discover --pattern "*_test.py" --start-directory src/test/py --top-level-directory .
+
+clean:
+	@echo "I'm cleaning OwO"
+	rm -rf venv
```

Comment:
It would be better to not pull the supporting directory out from underneath an activated `virtualenv`.
Please add a guard to detect if the `virtualenv` is active and deactivate it if so.

Something like:
```
declare -F deactivate && deactivate
rm -rf venv
```

`deactivate` is a shell function that is inserted into the environment when `source venv/bin/activate` is run and removed when `deactivate` is called.

Reply:
I've heard good things about `virtualenvwrapper`. Initially stuck with `virtualenv` since it's commonly used.

Other ones to look at before finalizing on that are the `pyenv` plugins for `virtualenv`s:
https://github.com/pyenv/pyenv-virtualenv
https://github.com/pyenv/pyenv-virtualenvwrapper

Reply:
Sounds good. I've cleared the clean-env and will work on the virtualenvwrapper on my own free time. Should be ready to get merged now.